src/frontend/src/lib/components/convert/ConvertInputAmount.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+	import { IconClose } from '@dfinity/gix-components';
+	import { debounce, isNullish, nonNullish } from '@dfinity/utils';
+	import { BigNumber } from '@ethersproject/bignumber';
+	import InputCurrency from '$lib/components/ui/InputCurrency.svelte';
+	import type { Token } from '$lib/types/token';
+	import { invalidAmount } from '$lib/utils/input.utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+
+	export let token: Token;
+	export let amount: number | undefined = undefined;
+	export let disabled: boolean | undefined = undefined;
+	export let customValidate: (userAmount: BigNumber | undefined) => void = () => {};
+	export let error: Error | undefined = undefined;
```

Comment:
This is maybe a new pattern? In which case will this be use?

Reply:
This is related to the above approach I described. Parent component passes it as `error={insufficientFundsError}`. This will set the input to error state only when provided amount is bigger than token balance. The input will stay in non-error state in case of `insufficientFundsForFee` is defined (the error message will be displayed above buttons).

Reply:
We are mixing some imperative and declarative approach, plus we are sharing up and down states, feels like we might end up with some desynchronization at some point. Is there another pattern that would allow to scope those errors within the component or transmit the information one way only with a single prop?

Reply:
I think it's now resolved with the changes to the approach I just pushed.

Reply:
Let's continue in the other thread.